### PR TITLE
Fix observability reporter - add URLs obfuscation

### PR DESCRIPTION
**Pull request by przlada**
I want to merge this change because it adds URL obfuscation to the observability reporter, as webhook target URLs can contain sensitive information.
For example:

- `awssqs://key:secret@sqs.us-east-2.amazonaws.com/xxxx/myqueue.fifo` -> `awssqs://key:***@sqs.us-east-2.amazonaws.com/xxxx/myqueue.fifo`

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Review by kadewu — APPROVED** [review]

File: saleor/webhook/observability/obfuscation.py [hunk 1]
```diff
@@ -47,6 +48,16 @@ def hide_sensitive_headers(
     }
 
 
+def obfuscate_url(url: str) -> str:
+    parts = urlparse(url)
+    if parts.username is None:
+        return url
```
Comment (NyanKiyoshi) [inline saleor/webhook/observability/obfuscation.py:54]: RFC 1738 says empty username is valid (it's not no user, but empty username)
```
>>> urlparse('https://:passwd@tst.com/foo').password
passwd
```

**Review by NyanKiyoshi — CHANGES_REQUESTED** [review]

File: saleor/webhook/observability/obfuscation.py (hunk 1 above)
Comment (przlada) [reply saleor/webhook/observability/obfuscation.py:54]: @NyanKiyoshi that is a great point, but this case is handled. If the user in URL is empty 
`urlparse("https://:passwd@tst.com/foo").username` returns an empty string, not `None`. So if username is `None`, that means there are no credentials in the URL. It is checked in the test case: 
```
"https://:password@example.com/test" -> "https://:{MASK}@example.com/test"
```

Comment (NyanKiyoshi) [reply saleor/webhook/observability/obfuscation.py:54]: Indeed, I missed this. Maybe you could add a comment above the line to make it more explicit?

**Review by NyanKiyoshi — APPROVED** [review]

File: saleor/webhook/observability/obfuscation.py (hunk 1 above)
Comment (przlada) [reply saleor/webhook/observability/obfuscation.py]: done 😄

**Review by IKarbowiak — APPROVED** [review]

**Pull request merged**